week1/homework/src/server.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 const http = require('http');
-
+const State = require('./state');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
 function createServer(port) {
-  let state = 10;
+  const state = new State();
+  function sendState(response) {
+    response.writeHead(200, { 'Content-Type': 'application/json' });
+    response.end(JSON.stringify(state));
```

Comment:
You are duplicating some of the same code for the same use,
this could be better in a matter of shortness, what about:
```javascript
// now this can be used in case of success or failer statuses, and you will have less lines
function sendJson(status, response, content) {
  response.writeHead(status, { 'Content-Type': 'application/json' });
  response.end(JSON.stringify(content));
}
```

Reply:
At first i deliberately isolated the error and state sending functions due to 'separation of concerns'. But as you mentioned with refactoring they are still very close concerns. No need to separate.
Refactored into  `sendJson`
Thanks for the tip